src/main/java/io/confluent/connect/elasticsearch/OffsetTracker.java

```diff
@@ -15,144 +15,26 @@
 
 package io.confluent.connect.elasticsearch;
 
-import org.apache.kafka.clients.consumer.OffsetAndMetadata;
-import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.connect.sink.SinkRecord;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.util.Collection;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.LinkedHashMap;
 import java.util.Map;
-import java.util.concurrent.atomic.AtomicLong;
-
-import static java.util.stream.Collectors.toMap;
-
-/**
- * Tracks processed records to calculate safe offsets to commit.
- *
- * <p>Since ElasticsearchClient can potentially process multiple batches asynchronously for the same
- * partition, if we don't want to wait for all in-flight batches at the end of the put call
- * (or flush/preCommit) we need to keep track of what's the highest offset that is safe to commit.
- * For now, we do that at the individual record level because batching is handled by BulkProcessor,
- * and we don't have control over grouping/ordering.
- */
-class OffsetTracker {
-
-  private static final Logger log = LoggerFactory.getLogger(OffsetTracker.class);
+import java.util.Set;
 
-  private final Map<TopicPartition, Map<Long, OffsetState>> offsetsByPartition = new HashMap<>();
-  private final Map<TopicPartition, Long> maxOffsetByPartition = new HashMap<>();
-
-  private final AtomicLong numEntries = new AtomicLong();
-
-  static class OffsetState {
-
-    private final long offset;
-    private volatile boolean processed;
-
-    OffsetState(long offset) {
-      this.offset = offset;
-    }
-
-    /**
-     * Marks the offset as processed (ready to report to preCommit)
-     */
-    public void markProcessed() {
-      processed = true;
-    }
-
-    public boolean isProcessed() {
-      return processed;
-    }
-  }
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.sink.SinkRecord;
 
-  /**
-   * Partitions are no longer owned, we should release all related resources.
-   * @param topicPartitions partitions to close
-   */
-  public synchronized void closePartitions(Collection<TopicPartition> topicPartitions) {
-    topicPartitions.forEach(tp -> {
-      Map<Long, OffsetState> offsets = offsetsByPartition.remove(tp);
-      if (offsets != null) {
-        numEntries.getAndAdd(-offsets.size());
-      }
-      maxOffsetByPartition.remove(tp);
-    });
-  }
+public interface OffsetTracker {
```

Comment:
As this is an interface we should document its purpose and each of its method. Could you please add some documentation?  You can copy it or move it from the implementation class